### PR TITLE
Replace HeterogeneousAtmosphere with HeterogeneousAtmosphereNew

### PR DESCRIPTION
**Pull request by nollety**
Resolves eradiate/eradiate-issues#105

# Description

Replace `HeterogeneousAtmosphere` with `HeterogeneousAtmosphereNew`.
`HeterogeneousAtmosphere` is renamed to `HeterogeneousAtmosphereLegacy`.

## Changes

- `HeterogeneousAtmosphere` renamed to `HeterogeneousAtmosphereLegacy`
  - corresponding `_heterogeneous.py` was renamed `_heterogeneous_legacy.py` 
  - corresponding `test_heterogeneous.py` renamed to `test_heterogeneous_legacy.py`
- `HeterogeneousAtmosphereNew` renamed to `HeterogeneousAtmosphere`
  - corresponding `_heterogeneous_new.py` renamed to `_heterogeneous.py`
  - corresponding `test_heterogeneous_new.py` renamed to `test_heterogeneous.py`

The diff does not make these changes clear, sorry about that... 😞 

## Notes

`OneDimSolverApp` now makes use of the new design for `HeterogeneousAtmosphere`.

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

**Comment by nollety** [comment]
@leroyvn It looks to me like our `HeterogeneousAtmosphereNew` (renamed `HeterogeneousAtmosphere`) makes a good job at replacing our old `HeterogeneousAtmosphere` (now `HeterogeneousAtmosphereLegacy`). 😃

File: docs/rst/reference/generated/factory/eradiate.scenes.atmosphere.atmosphere_factory.rst [hunk 1]
```diff
@@ -13,8 +13,8 @@ eradiate.scenes.atmosphere.atmosphere_factory
 
       * - ``heterogeneous``
         - :class:`HeterogeneousAtmosphere`
-      * - ``heterogeneous_new``
-        - :class:`HeterogeneousNewAtmosphere`
+      * - ``heterogeneous_legacy``
+        - :class:`HeterogeneousAtmosphereLegacy`
```
Comment (leroyvn) [inline docs/rst/reference/generated/factory/eradiate.scenes.atmosphere.atmosphere_factory.rst:17]: Out of curiosity: Did you edit this file manually?

File: docs/rst/user_guide/atmosphere/intro.rst [hunk 2]
```diff
@@ -96,7 +96,7 @@ Eradiate provides two atmosphere types:
   (:class:`~eradiate.scenes.atmosphere.HomogeneousAtmosphere`): radiative
   properties are uniform within the atmosphere.
 * heterogeneous atmosphere
-  (:class:`~eradiate.scenes.atmosphere.HeterogeneousAtmosphere`): radiative
+  (:class:`~eradiate.scenes.atmosphere.HeterogeneousAtmosphereLegacy`): radiative
```
Comment (leroyvn) [inline docs/rst/user_guide/atmosphere/intro.rst:99]: Side note: We'll have to update that when we'll finalise the docs for release.

File: eradiate/scenes/atmosphere/onedim_solver_app.rst [hunk 3]
```diff
@@ -0,0 +1,173 @@
+.. _sec-user_guide-onedim_solver_app:
```
Comment (leroyvn) [inline eradiate/scenes/atmosphere/onedim_solver_app.rst:1]: Is this file at its correct location?

File: docs/rst/reference/generated/factory/eradiate.scenes.atmosphere.atmosphere_factory.rst (hunk 1 above)
Comment (nollety) [reply docs/rst/reference/generated/factory/eradiate.scenes.atmosphere.atmosphere_factory.rst:17]: Yes, I did. Is that wrong?

File: docs/rst/user_guide/atmosphere/intro.rst (hunk 2 above)
Comment (nollety) [reply docs/rst/user_guide/atmosphere/intro.rst:99]: Yes, I agree. I decided to let the content of the tutorials and user guide utilize `HeterogeneousAtmosphereLegacy`, for simplicity.

File: eradiate/scenes/atmosphere/onedim_solver_app.rst (hunk 3 above)
Comment (nollety) [reply eradiate/scenes/atmosphere/onedim_solver_app.rst:1]: Probably not, I'll just delete it, I have no idea where that comes from.

**Comment by nollety** [comment]
Thanks for the feedback! @leroyvn 
If we could merge #124 first, that'd would nice because it's very small and the rebase here would be easier than the other way around

File: docs/rst/reference/generated/factory/eradiate.scenes.atmosphere.atmosphere_factory.rst (hunk 1 above)
Comment (leroyvn) [reply docs/rst/reference/generated/factory/eradiate.scenes.atmosphere.atmosphere_factory.rst:17]: It is generated by the `make docs-rst-api` target. I'll add a notice at the top of these files to make sure that people don't bother updating them manually.

**Comment by nollety** [comment]
@leroyvn Alright, all tests passed so I squashed the commits, this is ready to be merged! (don't forget #124 😉 )

**Comment by nollety** [comment]
@leroyvn ready for merging! 🚀

**Pull request merged**